src/main/java/eu/europeana/iiif/service/ManifestService.java

```diff
@@ -210,55 +215,121 @@ public String getRecordJson(String recordId, String wsKey, URL recordApiUrl) thr
+            return createAnnoPageMap(summary);
+        } else {
+            return null;
+        }
+    }
+
+    private Map<String, String[]> createAnnoPageMap(SummaryManifest summary){
+        LinkedHashMap<String, String[]> annoPageMap = new LinkedHashMap<>();
+        for (SummaryCanvas summaryCanvas : summary.getCanvases()){
+            List<String> annoPageIDs = new ArrayList<>();
+            List<SummaryAnnoPage> annoPageList = summaryCanvas.getAnnotations();
+            for (SummaryAnnoPage sap : annoPageList){
+                annoPageIDs.add(sap.getId());
+            }
+            annoPageMap.put(summaryCanvas.getPageNumber(), annoPageIDs.toArray(String[]::new));
+        }
+        return annoPageMap;
     }
 
+
+    /**
+     * Performs a HEAD request for a particular annotation page to see if the full text page exists or not
+     * @param fullTextUrl url to which HEAD request is sent
+     * @return true if it exists, false if it doesn't exists, null if we got no response
+     * @throws IIIFException when there is an error checking if a fulltext exists
+     */
+//    @HystrixCommand(commandProperties = {
+//            @HystrixProperty(name = "execution.isolation.thread.timeoutInMilliseconds", value = "5000"),
+//            @HystrixProperty(name = "fallback.enabled", value="true")
+//    }, fallbackMethod = "fallbackExistsFullText")
+//    Boolean existsFullTextOld(String fullTextUrl) throws IIIFException {
+//        Boolean result;
+//        try {
+//            try (CloseableHttpResponse response = headhttpClient.execute(new HttpHead(fullTextUrl))) {
+//                int responseCode = response.getStatusLine().getStatusCode();
+//                LOG.debug("Full-Text head request: {}, status code = {}", fullTextUrl, responseCode);
+//                if (responseCode == HttpStatus.SC_UNAUTHORIZED) {
+//                    throw new InvalidApiKeyException(APIKEY_NOT_VALID);
+//                } else if (responseCode == HttpStatus.SC_NOT_FOUND) {
+//                    result = Boolean.FALSE;
+//                } else if (responseCode == HttpStatus.SC_OK) {
+//                    result = Boolean.TRUE;
+//                } else {
+//                    // TODO when hysterix is enabled again, we can simply throw an error again
+//                    //throw new FullTextCheckException("Error checking if full text exists: "+response.getStatusLine().getReasonPhrase());
+//                    LOG.error("Error checking if full text exists: {}", response.getStatusLine().getReasonPhrase());
+//                    result = null;
+//                }
+//            }
+//        } catch (IOException e) {
+//            // TODO when hysterix is enabled again, we can simply throw an error again
+//            //throw new FullTextCheckException("Error checking if full text exists", e);
+//            LOG.error("Error checking if full text exists", e);
+//            result = null;
+//        }
+//        return result;
+//    }
```

Comment:
Can we delete this? Or do you prefer to wait until Hugo has approved and the current implementation is 'more final'?

Reply:
Deleted